app/models/payment.rb

```diff
@@ -0,0 +1,41 @@
+class Payment < ActiveRecord::Base
+  has_many :ticket_purchases
+  belongs_to :user
+  belongs_to :conference
+
+  attr_accessor :stripe_customer_email
+  attr_accessor :stripe_customer_token
+
+  validates :status, presence: true
+  validates :user_id, presence: true
+  validates :conference_id, presence: true
+
+  enum status: {
+    unpaid: 0,
+    success: 1,
+    failure: 2
+  }
+
+  def amount_to_pay
+    Ticket.total_price(conference, user, paid: false).cents
+  end
+
+  def purchase
+    gateway_response = Stripe::Charge.create source: stripe_customer_token,
+                                             receipt_email: stripe_customer_email,
+                                             description: "ticket purchases(#{user.username})",
+                                             amount: amount_to_pay,
+                                             currency: conference.tickets.first.price_currency
+
+    self.amount = gateway_response[:amount]
+    self.last4 = gateway_response[:source][:last4]
+    self.authorization_code = gateway_response[:id]
+    self.status = 'success'
+    true
+
+  rescue => error
```

Comment:
I recommend against rescuing from all types of errors.
For example, if we have some sort of exception from our own code (I dunno, division by zero or the Stripe gem changes its parameters, whatever), then users will see it as an error in their payment, and WE will never find out that something is wrong with our code.

It's better if we rescue from the exceptions that the Stripe gem can explicitly raise (invalid card? gateway error? timeout? ...?) and if there are any other errors, they will be sent to Errbit for analysis.